src/environment.js

```diff
@@ -1352,7 +1352,7 @@ export class Reference {
     );
     this.base = base;
     this.referencedName = refName;
-    invariant(!(refName instanceof AbstractValue) || !refName.mightNotBeString());
+    invariant(!(refName instanceof AbstractValue) || !(refName.mightNotBeString() && refName.mightNotBeNumber()));
```

Comment:
Seems more readable:
```js
!refName.mightNotBeString() || !refName.mightNotBeNumber()
```

Reply:
I specifically wrote the other way because I thought the other way is more readable. :-)